domain-driven design/ValueObjects/ValueObjects/ValueObjects/Result.cs

```diff
@@ -0,0 +1,13 @@
+namespace ValueObjects.ValueObjects;
+
+public record Result<T>
+{
+    public bool IsSuccess { get; private init; }
+    public T? Value { get; private init; }
+    public string? ErrorMessage { get; private init; }
+    
+    private Result(){}
```

Comment:
Do we need to declare this?

Reply:
Yes, because the default constructor is public and we only want this record to be instantiated via the Success or Failure static methods, thus we must make the default constructor private